client/protocols/restore.py

```diff
@@ -0,0 +1,116 @@
+from client.protocols.common import models
+from client.utils import auth
+from client.utils import network
+from datetime import datetime
+from urllib import error
+import logging
+import os
+import shutil
+
+log = logging.getLogger(__name__)
+
+GAE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+DEFAULT_BACKUP_DIR = "ok-backup"
+
+class RestoreProtocol(models.Protocol):
+    """Restores an assignment from an earlier backup."""
+
+    def run(self, messages):
+        """If the --restore flag is present, allow users to select
+        a backup to restore from.
+        """
+        if not self.args.restore:
+            return
+        
+        self.access_token = auth.authenticate(self.args.authenticate)
+        
+        print('Loading backups...')
+        
+        response = self.request('user')
+        if not response:
+            print('Could not connect to server')
+        user = response['data']['results'][0]
+        email = user['email'][0]
+        assign_id = self.get_assign_id(self.assignment.endpoint)
+        params = {
+            'assignment': assign_id   
+        }
+        backups = self.get_backups(email, assign_id)
+        current_time = datetime.now()
+        print('0: Cancel Restore')
+        for current in range(0, len(backups)):
+            backup = backups[current]
+            time_diff = get_time_diff(backup['timestamp'], current_time)
+            print('{0}: {1} by {2}'.format(current + 1, time_diff, backup['submitter']))
+        response = input('Backup #: ')
+        selection = int(response) - 1
+        if selection < 0 or selection >= len(backups):
+            return
```

Comment:
Should probably print out a message saying "Invalid option for restoring backups."